### PR TITLE
Update status check copy for clarity

### PR DESCRIPTION
**Pull request by jsdalton**
## Purpose

The first draft copy used for the status checks was a bit confusing and inconsistent in places. We want it to be very clear so the UX for service account installation is super clean.

There were also a few small edge case bugs and inconsistencies that need to be addressed. A lot of this compounded by the number of combinations, which makes it easy to over look edge cases.

A few bugs were found that are out of scope; tickets created.

## Approach

* Update the copy as needed. See screenshots for pertinent details

#### Before

image

#### After

image

#### Before

image

image

image

image

#### After

image

image

image

image

#### Before

image

image

image

#### After

image

image

image

### CODE REVIEW

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 1]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
+      icon: getSuccessIcon(''),
       title: CHECKLIST_NAMES.serviceAccount,
-      description: 'Success!',
+      description: 'Service account key is valid',
       disabled: false,
     },
     invalid: {
       icon: getErrorIcon(
-        'Failed to connect to your service account. Re-install your service account key if the problem persists'
+        `We couldn't connect to Google Analytics APIs using the service account key provided. Try re-installing your previous service account key or create and install a new key`
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:105]: i think the addition of `previous` here makes this weird. suggest removing or saying `current` instead – otherwise looks great

Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:105]: Agreed

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 2]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
       disabled: false,
     },
   },
   AdminApi: {
     success: {
-      icon: getSuccessIcon('Admin API successfully enabled'),
+      icon: getSuccessIcon(''),
       title: CHECKLIST_NAMES.adminApi,
-      description: 'Success!',
+      description: 'Analytics Admin API is enabled',
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:125]: is `Analytics Admin API` a proper noun? it's weird to me to say `Analytics Admin API` vs `Google Analytics Admin API` or just `Analytics admin API` (non-proper form)

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 3]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
+        'The Analytics Admin API allows Contentful to fetch the list of properties your service account has access to. Please enable this API inside your Google Cloud project to continue.'
+      ),
       title: CHECKLIST_NAMES.adminApi,
       description: 'Analytics Admin API is not yet enabled',
       disabled: false,
     },
     invalidServiceAccount: {
       icon: getClockIcon(
-        'Service account errors detected, please install a correctly configured service account installation'
+        'This check will not run until a valid service account key has been provided'
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:138]: 💯

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 4]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
       ),
       title: CHECKLIST_NAMES.adminApi,
-      description: 'Awaiting a correctly configured service account installation',
+      description: `Provide a valid service account key to run this check`,
       disabled: true,
     },
     error: {
-      icon: getErrorIcon('Failed to connect to the Admin API - did you enable the api?'),
+      icon: getErrorIcon(
+        'The Analytics Admin API allows Contentful to fetch the list of properties your service account has access to. Please enable this API inside your Google Cloud project.'
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:146]: here we're using `Analytics Admin API` as a proper noun again – is this correct?

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 2 above)
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:125]: I agree. Will fix in a few places where this is! (same for Data API)

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 5]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
+      icon: getSuccessIcon(''),
       title: CHECKLIST_NAMES.dataApi,
-      description: 'Success!',
+      description: 'Analytics Data API is enabled',
       disabled: false,
     },
     firstTimeSetup: {
-      icon: getArrowIcon('Please enable the Data API to run this check'),
+      icon: getArrowIcon(
+        'The Analytics Data API allows Contentful to fetch analytics data from the property you specify. Please enable this API inside your Google Cloud project to continue.'
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:162]: a little bit of divergence here from the one above – let's end them both the same: `Please enable this API inside your Google Cloud project.`

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 6]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
       ),
       title: CHECKLIST_NAMES.dataApi,
-      description: 'Awaiting a correctly configured service account installation',
+      description: `Provide a valid service account key to run this check`,
       disabled: true,
     },
     error: {
-      icon: getErrorIcon('Failed to connect to the Data API - did you enable the api?'),
+      icon: getErrorIcon(
+        'The Analytics Data API allows Contentful to fetch analytics data from the property you specify. Please enable this API inside your Google Cloud project.'
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:178]: suggest rewording `inside` [your GC project] to `in` [your GC project]

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 5 above)
Comment (ghepting) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:162]: also i think `in` is better than `inside` (mentioned this below, too)

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 4 above)
Comment (ghepting) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:146]: also i think `in` is better than `inside` (mentioned this below, too)

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 7]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
   },
   GA4Properties: {
     success: {
-      icon: getSuccessIcon(
-        'Your service account has access to assign a Google Analytics 4 property'
-      ),
+      icon: getSuccessIcon(''),
       title: CHECKLIST_NAMES.ga4Properties,
-      description: 'Success!',
+      description: 'Service account has "viewer" access',
```
[the remark this answers is not in the document]
Comment (ghepting) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:189]: ProTip: use `Option-Shift-]` on OSX to type these while you're coding 🙂

[the remark this answers is not in the document]
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:189]: Brave!

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 8]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
       disabled: false,
       checklistUrl: {
         title: 'Details',
         url: 'https://analytics.google.com/analytics/web/',
       },
     },
     firstTimeSetup: {
-      icon: getClockIcon(`Check will run once the ${CHECKLIST_NAMES.adminApi} is enabled`),
+      icon: getArrowIcon(
+        `You need to grant viewer access to your service account in a Google Analytics 4 property.`
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:198]: should we smart quote `viewer` here, too so we're consistent?

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 9]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
+      ),
+      title: CHECKLIST_NAMES.ga4Properties,
+      description: "Service account doesn't have access to any GA4 properties",
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:201]: is `GA4` slang? Should we say the full name `Google Analytics 4 properties` here or does it not fit?

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 4 above)
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx]: Agreed x2

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 10]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
+      disabled: false,
+      checklistUrl: {
+        title: 'Grant access',
+        url: 'https://analytics.google.com/analytics/web/',
+      },
+    },
+    firstTimeSetupNotEnabled: {
+      icon: getClockIcon(
+        `You'll need to grant viewer access to your service account in a Google Analytics 4 property. This check will run once the ${CHECKLIST_NAMES.adminApi} has been enabled.`
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:210]: `”viewer”` here too

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 11]
```diff
@@ -93,113 +95,134 @@ export type ChecklistStatus = {
...
+      ),
       title: CHECKLIST_NAMES.ga4Properties,
       description: `Enable Analytics Admin API to run this check`,
       disabled: true,
     },
     invalidServiceAccount: {
       icon: getClockIcon(
-        'Service account errors detected, please install a correctly configured service account installation'
+        'This check will not run until a valid service account key has been provided'
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:218]: man we should really store this as a variable. way too much repeated use of this to keep copy/pasting it everywhere i think

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 5 above)
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx]: > a little bit of divergence here from the one above – let's end them both the same: Please enable this API inside your Google Cloud project.

Divergence here is intentional. For first time setup we're emphasizing that you need to do this to continue the process. For the second you just need to enable it cuz somehow it got disabled.

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 12]
```diff
@@ -209,17 +232,19 @@ export const CHECKLIST_STATUSES: ChecklistStatus = {
         'The service account must have properties assigned to the account for this check to pass'
       ),
       title: CHECKLIST_NAMES.ga4Properties,
-      description: 'Service account failed to access an Analytics property',
+      description: "Service account doesn't have access to any GA4 properties",
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:237]: `GA4` usage here, too  – recommend avoiding shorthand if UI/visual issues do not result in saying `Google Analytics 4 properties`

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx [hunk 13]
```diff
@@ -299,11 +324,17 @@ export const getGa4PropertyErrorChecklistStatus = (
   adminApiError: ApiErrorType | undefined,
   ga4PropertiesError: ApiErrorType | undefined
 ): ChecklistRow => {
+  console.log('isFirstSetup', isFirstSetup);
+  console.log('invalidServiceAccountError', invalidServiceAccountError);
+  console.log('adminApiError', adminApiError);
+  console.log('ga4PropertiesError', ga4PropertiesError);
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:332]: some debugging logs leftover here FYI

- [ ] remove debugging console.logs

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 9 above)
Comment (ryunsong-contentful) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:201]: If it's too long maybe just say `analytics properties`?

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.spec.tsx [hunk 14]
```diff
@@ -38,6 +38,6 @@ describe('Installed Service Account Card', () => {
     });
 
     await screen.findByText('Google Service Account Details');
-    await screen.findByText('Active');
+    await screen.findByText('Access correctly configured');
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.spec.tsx:41]: this feels a bit weird to me... WDYT about sth like `Configuration verified` or `Access verified`?

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 9 above)
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx:201]: Yeah agreed. We're short on space in this section so reworded as follows:

>  Service account doesn't have access to a Google Analytics 4 property

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx [hunk 15]
```diff
@@ -222,9 +223,9 @@ const DisplayServiceAccountCard = (props: Props) => {
       ga4PropertiesError;
     if (configError) {
       return isFirstSetup ? (
-        <Badge variant="warning">Incomplete</Badge>
+        <Badge variant="primary">Finish configuration steps below</Badge>
       ) : (
-        <Badge variant="negative">"Some checks were unsuccessful"</Badge>
+        <Badge variant="negative">Problems with your configuration</Badge>
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx:228]: let's remove `your` here and try to stop using that pronoun. WDYT about just `Problem with configuration` or even `Problem with configuration detected`?

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx [hunk 16]
```diff
@@ -237,15 +238,17 @@ const DisplayServiceAccountCard = (props: Props) => {
       );
     }
 
-    return <Badge variant="positive">Active</Badge>;
+    return <Badge variant="positive">Access correctly configured</Badge>;
```
Comment (ghepting) [inline apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx:241]: whatever you do about my [comment above](https://github.com/contentful/apps/pull/3012/files#r1153818764) – the same applies to this spot, too. perhaps another variable. i almost want to put all of this copy into a JSON object at this point and treat it like we would i18n strings so we're DRY about it more easily

**Review by ghepting — APPROVED** [review]
left a bunch of callouts and suggestions, but is way better, thank you @jsdalton!

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx (hunk 16 above)
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx:241]: I know that feel 😅 . For now I avoided refactoring entirely since there's a lot more that needs to be done where that came from!

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx (hunk 15 above)
Comment (jsdalton) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx:228]: This cleans up the 3 possibilties to:

**Status:**

* <Badge variant="primary">Finish configuration steps below</Badge>
* <Badge variant="negative">Problems with configuration</Badge>
* <Badge variant="positive">Successfully configured</Badge>

**Review by ghepting — APPROVED** [review]

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx (hunk 15 above)
Comment (ghepting) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/DisplayServiceAccountCard.tsx]: 👍 sgtm

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 9 above)
Comment (ghepting) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx]: perfect.

File: apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx (hunk 5 above)
Comment (ghepting) [reply apps/google-analytics-4/frontend/src/components/config-screen/api-access/display/ChecklistUtils.tsx]: ah, gotcha, thanks for clarifying. carry on

**Review by lilbitner — APPROVED** [review]
Looks great to me ✨
